### PR TITLE
Allow inline (trailing) comments in INI files

### PR DESCRIPTION
**Pull request by rjungheinrich**
Inline or trailing comments as these ones:
```
[Foo] # This is an inline comment
bar=Hello world! # This is another inline comment
```
are a handy feature often requested by end clients.

Although there is [no standard on this](https://en.wikipedia.org/wiki/INI_file#Comments), I think they would be a good feature for the library.

This pull request implements such a feature.

### CODE REVIEW

**Comment by Rookfighter** [comment]
Hi,
thanks for your pull request. You are right that inline comments are a common feature. I intentionally did not implement inline comments at the time, because it was more effort and I did not need the feature.

There is one issue with your changes: I believe it is now not possible anymore to use the hash symbol "#" in any string in a ini field.
E.g.

```
[Foo]
myVal=Hello # World
```

```myVal``` will then only contain ```Hello ```, instead of ```Hello # World```. This can also lead to unexpected behaviour. Please verify this issue with tests. Also, introduce some way to use the hash symbols in strings (e.g. escaping as ```\#```)

**Comment by rjungheinrich** [comment]
You are right. I hadn't realized that this change is not 100% retro-compatible since in certain corner cases, like the one you've shown, we are changing behavior.

I will definetively add a test that will show that.

The suggestion for adding a way to escape the comment prefix is a very good one, I will work on that. Do you thing that the escape character should be configurable?

To clarify, with this ini file:
```
[Foo]
one=Hello # World
two=Hello \# World
three=Hello \World
```

We will end up with:

- one containing "Hello"
- two containing "Hello # World"
- three containing "Hello \World"

**Comment by rjungheinrich** [comment]
Modification to the pull request done.

I'd appreciate it if you'd review it when you have time.

File: include/inicpp.h [hunk 1]
```diff
@@ -207,12 +208,29 @@ namespace ini
                 str = "";
         }
 
+        void erase_comment(std::string &str)
```
Comment (Rookfighter) [inline include/inicpp.h:211]: Please change the function name to Camel-Case to match the variable naming style.

Comment (rjungheinrich) [reply include/inicpp.h]: Sure. Fixed in the following commit.
